### PR TITLE
#2733 Added new functions to support Keptn Smart Auth

### PR DESCRIPTION
**Pull request by ankitjain28may**
Signed-off-by: ankitjain28may <ankitjain28may77@gmail.com>

### CODE REVIEW

**Comment by ankitjain28may** [comment]
[https://github.com/keptn/keptn/pull/2733](https://github.com/keptn/keptn/pull/2733)

File: pkg/kubernetes-utils.go [hunk 1]
```diff
@@ -295,6 +304,92 @@ func ExistsNamespace(useInClusterConfig bool, namespace string) (bool, error) {
...
+	if err != nil {
+		return "", err
+	}
+	keptnService, err = clientset.CoreV1().Services(namespace).Get(serviceName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	switch keptnService.Spec.Type {
+	case "LoadBalancer":
+		return keptnService.Status.LoadBalancer.Ingress[0].IP, nil
```
Comment (bacherfl) [inline pkg/kubernetes-utils.go:370]: please add a check for the length of the `keptnService.Status.LoadBalancer.Ingress` array

Comment (ankitjain28may) [reply pkg/kubernetes-utils.go]: Done. Please review

**Review by bacherfl — APPROVED** [review]
lgtm

**Pull request merged**